contracts/volt/VoltV2.sol

```diff
@@ -0,0 +1,426 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+// Forked from Uniswap's UNI
+// Reference: https://etherscan.io/address/0x1f9840a85d5af5bf1d1762f925bdaddc4201f984#code
+
+contract VoltV2 is Ownable {
+    using SafeCast for *;
+
+    /// @notice EIP-20 token name for this token
+    // solhint-disable-next-line const-name-snakecase
+    string public constant name = "Volt";
+
+    /// @notice EIP-20 token symbol for this token
+    // solhint-disable-next-line const-name-snakecase
+    string public constant symbol = "VOLT";
+
+    /// @notice EIP-20 token decimals for this token
+    // solhint-disable-next-line const-name-snakecase
+    uint8 public constant decimals = 18;
+
+    /// @notice Total number of tokens in circulation
+    // solhint-disable-next-line const-name-snakecase
+    uint256 public totalSupply;
+
+    /// @notice Allowance amounts on behalf of others
+    mapping(address => mapping(address => uint256)) internal allowances;
+
+    /// @notice Official record of token balances for each account
+    mapping(address => uint256) internal balances;
+
+    /// @notice A record of each accounts delegate
+    mapping(address => address) public delegates;
+
+    /// @notice A checkpoint for marking number of votes from a given block
+    struct Checkpoint {
+        uint32 fromBlock;
+        uint256 votes;
```

Comment:
did we change this? it feels like it would be better to pack this variable into a single storage slot, so make votes be uint224 so that we save a whole SSTORE on each transfer.

Reply:
Yeah we changed it from a `uint96` to a `uint256`, as we changed the balances by the same amount, I think we should change this if we are changing the balances etc to a `uint224`  as the votes and the balances should align

Reply:
I really don't like this as it means transferring this new token will be 21k gas more than it should be. Let's jump on a call when you're free and talk over this + ACL

Reply:
using `uint224` for balances and votes
<img width="509" alt="uint224" src="https://user-images.githubusercontent.com/32230412/194108656-b06b8770-f2d2-4677-ac63-f05872d71673.png">

using `uint256` for balances and votes
<img width="511" alt="uint256" src="https://user-images.githubusercontent.com/32230412/194108589-5ed50b30-dadf-439f-86c2-88e2a2a839f5.png">